modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -280,11 +280,11 @@ public function order_process( $order_id ) {
 
 		// loop through products in the order and queue a purchase event.
 		foreach ( $order->get_items() as $order_item_id => $order_item ) {
-			$product = $order->get_product_from_item( $order_item );
+			$product = is_callable( array( $order_item, 'get_product' ) ) ? $order_item->get_product() : null;
```

Comment:
I can't be certain that `$order_item` is `WC_Order_Item_Product`, so I'm doing an is_callable check here (i.e. defensive/paranoid code). 

I don't know how to test the non-product order item case; I think if there may be woo extensions that add other kinds of line items. @KokkieH let me know you have ideas for how to test this with more interesting orders! (And - thanks for reporting this!)

Reply:
I'm afraid I don't know how to test it, period :)

I wasn't able to replicate the issue myself, and the users in the forum thread didn't provide any more detailed information than "it happens on the checkout page".

Someone from WooCommerce might have an idea, though.